Consolidated KYC Match API definition

ToshiWakayama-KDDI opened the pull request:
#### What type of PR is this?

* enhancement/feature

#### What this PR does / why we need it:

* Provide a consolidated KYC Match API definition based on subproject discussion for the initial version.

ToshiWakayama-KDDI committed:
Add files via upload

EfthymisIsaakidis-DTCS commented:
Hello Colleagues,
Regarding parameter naming convention, we (DT group) believe it would be better to use the same notation with GSMA Mobile Connect KYC Match, in other words "snake notation" (thus use underscore "_") instead of "camel notation". This will lead to a uniform approach more compatible with that of GSMA. 
For example:
|**attribute**|**Match Request**|**Match Response**|
|**e.g. given name** | given_name | given_name_match |

EfthymisIsaakidis-DTCS commented:
Hello Colleagues,
Regarding the match result response specifications, we (DT group) believe it would be better to define and use the same values with GSMA Mobile Connect KYC Match:
“Y”– match is successful, “N-NA” - match failed, data is not available, “N-AV” – match failed; data is available, “N-AD” – match failed, data is available, but access is denied.
This will lead to a uniform approach and will also cover the "access denied" scenario.

GillesInnov35 commented:
@EfthymisIsaakidis-DTCS, I'm sorry but I don't agree with your proposition because as it is for all CAMARA API design, we must be compliant with Camara commonalities guidelines
https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md

at **§4.2 Input/Output Resource** Definition it is mentionned that _**API input and output business data will follow the camelCase notation.**  and "_" is never used in the other API field notation.
thanks 
BR
Gilles

fernandopradocabrillo commented:
Agree with @GillesInnov35

We have to follow CAMARA guidelines as it is done in the rest of the API families.

ToshiWakayama-KDDI commented:
Hi @EfthymisIsaakidis-DTCS, 
Thanks for your comment about parameter naming, but we have already agreed to follow the camelCase notation.  As others mentioned, we have to follow Camara commonalities guidelines.

Thank you for your understanding.

Best regards,
Toshi

ToshiWakayama-KDDI commented:
Hi @EfthymisIsaakidis-DTCS ,
Thank you for your comments about match result response, but I have to advise you that we have been discussing this long, and we agreed at the meeting on Tuesday 12th December, as below, which is also described in Issue #21. 

- **true: successful match**
- **false: failed match**
- **not_available: data is not available**

I forgot to update Issue #21, which may confuse you, so I will update it shortly as it was agreed on 12th December.

So, @EfthymisIsaakidis-DTCS , sorry, but if you propose the modification, please do that for our future releases.

Thank you for your cooperation.

Best regards,
Toshi

EfthymisIsaakidis-DTCS commented:
Hello Colleagues,
Thank you for your feedback on parameter naming convention. Considering your comments and the Section 4.2 in Commonalities we have no objections to follow your suggestion on this issue.

Best Regards,
Efthymis

EfthymisIsaakidis-DTCS commented:
Hello @ToshiWakayama-KDDI ,

Thank you for your feedback about match result response. I acknowledge that my comment comes somehow late, but I do think that here we are taking a decision that we may need to change in the future . The restriction of access in some of the user's info attributes may be needed in some use cases depending on different laws and regulations in different countries as well. GSMA specs have considered this. Of course I agree that we proceed as already agreed and discuss this modification in a future release, if and when it is needed, but it may be more difficult and "costly" to do the change later.

BR,
Efthymis

ToshiWakayama-KDDI committed:
Update kyc-match.yaml

Update based on the outcome at 2023/12/12 meeting

ToshiWakayama-KDDI commented:
Hi @EfthymisIsaakidis-DTCS ,

Thank you, but we have agreed the match result response for our initial version, and we have to complete our initial YAML by Tuesday 19th.  If we restart match result response discussion, it will take more time and never end.  Especially you mentioned the user's info attributes, and we decided to delay user's info attributes discussion till  future releases.  I understand your point, but why don't we complete our first version once as a basis, and then I think we can modify and refine it in an agile fashion to make it better.  As you know, if we look at some other APIs in CAMARA, they have had several update versions.  This is my current view.

Best regards,
Toshi

ToshiWakayama-KDDI committed:
Update kyc-match.yaml

Error Responses modified as agreed in Issue #24

ToshiWakayama-KDDI commented:
> Hi @EfthymisIsaakidis-DTCS , Thank you for your comments about match result response, but I have to advise you that we have been discussing this long, and we agreed at the meeting on Tuesday 12th December, as below, which is also described in Issue #21.
> 
> * **true: successful match**
> * **false: failed match**
> * **not_available: data is not available**
> 
> I forgot to update Issue #21, which may confuse you, so I will update it shortly as it was agreed on 12th December.
> 
> So, @EfthymisIsaakidis-DTCS , sorry, but if you propose the modification, please do that for our future releases.
> 
> Thank you for your cooperation.
> 
> Best regards, Toshi

Hi @EfthymisIsaakidis-DTCS,
Regarding your previous question on Match Response, I have an additional comment.  I think  our current KYC Match can cover your scenarios (MC scenarios?) as below: 
- Y - true
- N-NA - not available
- N-AV - false
- N-AD - can be covered by Error Response 401 or 403

Thanks.
Toshi

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+
+    apiKey:
+      type: apiKey
+      in: header
+      name: X-KDDI-API-KEY
+      description: API key given to each client, to be used for API-GW authentication.
+    Authorization:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+      description: Access toke given by IdP(authentication)
```
We can remove this schemas, they are not covered in CAMARA

```suggestion
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+
+        - Indicated param combination is invalid (`"code": "KNOW_YOUR_CUSTOMER.INVALID_PARAM_COMBINATION","message": "Indicated parameter combination is invalid"`)
+      content:
+        application/json;charset=UTF-8:
```
According to CAMARA guidelines:

```suggestion
        application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+        - openId:
+          - know-your-customer:match
+
+        - apiKey: []
+          Authorization: []
```
```suggestion
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+                message: sorry_error
+
+    Generic504:
+      description: Request time exceeded. If it happens repeatedly, consider reducing the request complexity
```
```suggestion
      description: Request time exceeded. If it happens repeatedly, consider reducing the request complexity
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+                message: Server error
+
+    Generic503:
+      description: Service unavailable. Typically the server is down.  Problem with MNO's server side.  Any unexpected error within MNO's servers.
```
```suggestion
      description: Service unavailable. Typically the server is down.  Problem with MNO's server side.  Any unexpected error within MNO's servers.
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+                message: not_found_contractor/not_found
+
+    Generic500:
+      description: Server error.  Problem with MNO's server side.  Some processing error within MNO's servers.
```
```suggestion
      description: Server error.  Problem with MNO's server side.  Some processing error within MNO's servers.
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+                message: The idDocument needs to match the one associated with the provided phoneNumber
+
+    Generic404:
+      description: |
+        Not Found error.  Error if URL is wrong / user is not found.
```
```suggestion
      description: |
        Not Found error.  Error if URL is wrong / user is not found.
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+        - Phone number cannot be deducted from access token context.(`{"code": "KNOW_YOUR_CUSTOMER.INVALID_TOKEN_CONTEXT","message": "Phone number mismatch with access token context"}`)
+        - The idDocument property is missing.(`{"code": "KNOW_YOUR_CUSTOMER.ID_DOCUMENT_REQUIRED","message": "The idDocument is required to perform the properties validation"}`)
+        - The idDocument does not match the one associated to the provided phoneNumber in the OB systems.(`{"code": "KNOW_YOUR_CUSTOMER.ID_DOCUMENT_MISMATCH","message": "The idDocument needs to match the one associated with the provided phoneNumber"}`)
+      content:
```
```suggestion
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
      content:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+                message: Indicated parameter combination is invalid
+
+    Generic401:
+      description: Authentication problem with the client request. Unauthorized error.  Access Token related errors.
```
```suggestion
      description: Authentication problem with the client request. Unauthorized error.  Access Token related errors.
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+        In addition to regular scenario of `INVALID_ARGUMENT`, another scenario may exist.
+
+        - Indicated param combination is invalid (`"code": "KNOW_YOUR_CUSTOMER.INVALID_PARAM_COMBINATION","message": "Indicated parameter combination is invalid"`)
+      content:
```
```suggestion
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
      content:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+
+      responses:
+        '200':
+          description: OK
```
```suggestion
          description: OK
          headers:
            x-correlator:
              $ref: '#/components/headers/x-correlator'
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+        description: Base path for the KYC-Match API
+
+tags:
+  - name: KYC_Match
```
I think that adding KYC is a bit redundant

```suggestion
  - name: Match

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+        - apiKey: []
+          Authorization: []
+
+      requestBody:
```
```suggestion
      parameters:
        - $ref: '#/components/parameters/x-correlator'
        
      requestBody:
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+      bearerFormat: JWT
+      description: Access toke given by IdP(authentication)
+
+  schemas:
```
```suggestion
  headers:
    x-correlator:
      description: Correlation id for the different services
      schema:
        type: string
   
   parameters:
    x-correlator:
      name: x-correlator
      in: header
      description: Correlation id for the different services
      schema:
        type: string
        
  schemas:
```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+    Generic401:
+      description: Authentication problem with the client request. Unauthorized error.  Access Token related errors.
+      content:
+        application/json;charset=UTF-8:
```
```suggestion
        application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+        - The idDocument property is missing.(`{"code": "KNOW_YOUR_CUSTOMER.ID_DOCUMENT_REQUIRED","message": "The idDocument is required to perform the properties validation"}`)
+        - The idDocument does not match the one associated to the provided phoneNumber in the OB systems.(`{"code": "KNOW_YOUR_CUSTOMER.ID_DOCUMENT_MISMATCH","message": "The idDocument needs to match the one associated with the provided phoneNumber"}`)
+      content:
+        application/json;charset=UTF-8:
```
```suggestion
        application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+      description: |
+        Not Found error.  Error if URL is wrong / user is not found.
+      content:
+        application/json;charset=UTF-8:
```
```suggestion
        application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+    Generic500:
+      description: Server error.  Problem with MNO's server side.  Some processing error within MNO's servers.
+      content:
+        application/json;charset=UTF-8:
```
```suggestion
        application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+    Generic503:
+      description: Service unavailable. Typically the server is down.  Problem with MNO's server side.  Any unexpected error within MNO's servers.
+      content:
+        application/json;charset=UTF-8:
```
```suggestion
        application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+    Generic504:
+      description: Request time exceeded. If it happens repeatedly, consider reducing the request complexity
+      content:
+        application/json;charset=UTF-8:
```
```suggestion
        application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+          examples:
+            NotFound:
+              value:
+                status: 403
```
```suggestion
                status: 404

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+                status: 503
+                code: UNAVAILABLE
+                message: Service unavailable
+            Sorry_error:
+              value:
+                status: 503
+                code: UNAVAILABLE
+                message: sorry_error
```
I think we can remove this example, is not giving much information

```suggestion
            Sorry_error:
              value:
                status: 503
                code: UNAVAILABLE
                message: sorry_error

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+      requestBody:
+        required: true
+        content:
+          application/json;charset=UTF-8:
```
```suggestion
          application/json:

```

GillesInnov35 commented:
@ToshiWakayama-KDDI , thanks a lot for your work. For information it seems that at line 296 there's a little mistake. A **'** is missing  after 'not_available 
Do you confirm ?
BR 
Gilles

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,536 @@
+
+        address:
+          type: string
+          description: Address of the customer
```
I think we should give a more detailed description here. For example, do we allow to input the type of street (Street, Avenue, Platz, etc? do we add the locality/region/country, etc?

```suggestion
          description: Address of the customer. This field can be the combination of multiple partial address parameters as streetName and streetNumber. It should not include street type.

```

fernandopradocabrillo reviewed (commented):
@ToshiWakayama-KDDI 
A few comments and clarifications. They seem a lot but most of them are repeated in several places 😄

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Add ' to bottom of not_available for middleNamesMatch.  It seems Toshi accidentally deleted it.

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposed to remove KDDI specific part regarding apiKey/authorization.

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Status: was 403 (wrongly), corrected to 404; Fernando pointed out.

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

apikey [] and authorization [] removed; Fernando pointed out.

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Remove Error 503, UNAVAILABLE, sorry_error as it is not giving much information; Fernando proposed

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI commented:
HIi @fernandopradocabrillo ,

> description: Address of the customer. This field can be the combination of multiple partial address parameters as streetName and streetNumber. It should not include street type.

For my clarification, some MNOs including KDDI use the Address field as one address string, but you prefer to use this as the combination of partial address parameters, so, you want to add that your case is possible.  Is this correct understanding?

As we do not have the 'street type' attribute, there is no need for "It should not include street type", I think.

Many thanks,
Toshi

fernandopradocabrillo commented:
Hi @ToshiWakayama-KDDI 
> (1) Change 'application/json;charset=utf-8' to 'application/json' I think that there is a recent commonalities discussion (Issue #112) related this point. It is stated that, although utf-8 is the default character set, adding 'charset=utf-8' is optionally ok. But as it is still ongoing and not concluded, I would like to add 'charset=utf-8' just to make sure.
> 
> Are you uncomfortable with this (adding charset=utf-8)?

I wasn't aware of that issue in commonalities! It seems that the comments goes in the line of not including the carset unless it's something different from UTC-8. I think it is still a bit early to include it, it doesn't seem matured enough (in my opinion).

> (2) x-correlator in header I have checked the CAMARA API Desing Guidelines, but I am not quite sure if 'x-correlator' is mandatory and if it is required for KYC Match and KYC Fill-in APIs. So, I prefer to not include 'x-correlator'. Do you think we need to include it?

There has been multiple issues in commonalities regarding the x-correlator topic, this is the last one: https://github.com/camaraproject/Commonalities/issues/101

Eric from Vodafone was the one that raised the doubt about including it or not and, as per his last comment, he is finally inclined to include it. If you take a look, it is already in most of the CAMARA APIs, I think only Device Status and Sim swap still don't have it, but at least in the latter a PR is already created. So the only thing left is to adapt the guidelines, which I think will be done shortly.

> For my clarification, some MNOs including KDDI use the Address field as one address string, but you prefer to use this as the combination of partial address parameters, so, you want to add that your case is possible. Is this correct understanding?

No, I didn't explain myself correctly. I think that this field is similar to the given name parameter. In essence it is the combination of other partial parameters. What is an address composed of? A street name, a street number, a locality, a region, a country, a postal code, etc I think we should define at least a very basic rules of normalization, for example not including the street_type (street, avenue, park, square, etc)

ToshiWakayama-KDDI commented:
Hi @fernandopradocabrillo, 

Thank you for your reply.

> I wasn't aware of that issue in commonalities! It seems that the comments goes in the line of not including the charset unless it's something different from UTF-8. I think it is still a bit early to include it, it doesn't seem matured enough (in my opinion).

So, are you saying that you are not comfortable with adding 'charset=utf-8', even though we should use UTF-8 as the default characterset?  If so, it is OK to remove 'charset=utf-8'.


> There has been multiple issues in commonalities regarding the x-correlator topic, this is the last one: https://github.com/camaraproject/Commonalities/issues/101
> 
> Eric from Vodafone was the one that raised the doubt about including it or not and, as per his last comment, he is finally inclined to include it. If you take a look, it is already in most of the CAMARA APIs, I think only Device Status and Sim swap still don't have it, but at least in the latter a PR is already created. So the only thing left is to adapt the guidelines, which I think will be done shortly.

Thanks.  I will check Commonalities Issue #101.  In the meantime, it is OK for me to include the x-correlator.  Do in Roma, as the Romans do.  But if I find something different, please let us revisit this issue.


> No, I didn't explain myself correctly. I think that this field is similar to the given name parameter. In essence it is the combination of other partial parameters. What is an address composed of? A street name, a street number, a locality, a region, a country, a postal code, etc I think we should define at least a very basic rules of normalization, for example not including the street_type (street, avenue, park, square, etc)

We have not discussed normalization yet, so I do not think I can agree with it.  
Maybe we could add like this:
 description: Address of the customer. This field can be used for the whole address but also can be the combination of multiple partial address parameters as streetName and streetNumber.

What do you think?

Thanks.

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposasl: to add headers / x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add parameters/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to add headers/ x-correlator and parameters/ x-correlator

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update kyc-match.yaml

Remove error

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (dismissed):
OK, thanks

ToshiWakayama-KDDI committed:
Update kyc-match.yaml

Remove error

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (approved):
OK, thanks

GillesInnov35 reviewed (dismissed):
OK, thanks

ToshiWakayama-KDDI commented:
Hi @fernandopradocabrillo , Hi @GillesInnov35 

I have committed all the change proposals created by Fernando except description of Address of the customer.  I think the proposal for desrciption of Address of the customer is related to normalisation, so it should be delayed to future discussion.  

And I have corrected some errors detected by the Swagger Editor.

So, I think the Match YAML proposal can be merged now.

Many thanks,
Toshi

fernandopradocabrillo commented:
Hi @ToshiWakayama-KDDI 
> So, are you saying that you are not comfortable with adding 'charset=utf-8', even though we should use UTF-8 as the default characterset? If so, it is OK to remove 'charset=utf-8'.

Yes, I prefer to not explicitly include it yet BUT I'm of course open to add it later if it is decided in commonalities. For now we can just asume it is utf-8 like in the rest of the APIs. I'm checking internally if our system supports including the charset explicitly at the moment becasue I'm not sure.

> We have not discussed normalization yet, so I do not think I can agree with it. Maybe we could add like this: description: Address of the customer. This field can be used for the whole address but also can be the combination of multiple partial address parameters as streetName and streetNumber.
> 
> What do you think?

Yes, I'm okey with this description. The street_type thing was just to make the usecase a little bit more specific so the client knows what to send. As "4 Privet Drive" is not the same as "4 Privet Drive Street". But I'm okey discussing it later.

Thanks!

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,573 @@
+              $ref: '#/components/headers/x-correlator'
+
+          content:
+            application/json;charset=UTF-8:
```
```suggestion
            application/json:

```

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,573 @@
+      properties:
+        phoneNumber:
+          type: string
+          description: A phone number belonging to the user in E.164 format (starting with country code). Optionally prefixed with '+'.
```
An this one just to align with the commonalities Glossary

```suggestion
          description: A public identifier addressing a telephone subscription. In mobile networks it corresponds to the MSISDN (Mobile Station International Subscriber Directory Number). In order to be globally unique it has to be formatted in international format, according to E.164 standard, optionally prefixed with '+'.

```

ToshiWakayama-KDDI replied:
> An this one just to align with the commonalities Glossary

Thank you very much, but never ending.  Could we make this the last change, please?  GSMA is waiting.  We can do corrections later.

ToshiWakayama-KDDI commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -0,0 +1,573 @@
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+  version: 0.1.0
```
> I guess it showed an error in the linter but we have to keep it as wip until we are ready to do a -rc version

I do not unerstand why we have to keep it as wip, but if Telefonica needs it, I am ok.

fernandopradocabrillo replied:
It is not a Telefónica request, it's the WoW for making a release in CAMARA

fernandopradocabrillo replied:
hi @GillesInnov35 

I don't find the particular documentation but this is how it's been made in the rest of the WG.
-wip: for ongoing version being developed
-rc: once there is an agreement that the scope has been fulfilled, the version changes for a -rc

If no further issues has been raised and there is nothing to fix, then a release branch (release-0.1.0) needs to be made, removing the -rc, along with the CHANGELOG.md and the version tag (v0.1.0 in this case) Documentation for this process [here](https://github.com/camaraproject/Commonalities/blob/main/documentation/Camara_Versioning_Guidelines.md)

A few examples in other CAMARA WG: 
- Device Status: https://github.com/camaraproject/DeviceStatus/pull/95#discussion_r1430044237
- Device Location: https://github.com/camaraproject/DeviceLocation/blob/main/code/API_definitions/location-verification.yaml#L52
- QoD: https://github.com/camaraproject/QualityOnDemand/blob/main/code/API_definitions/qod-api.yaml#L69

GillesInnov35 replied:
ok thanks.

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to remove charaset=UTF-8

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI committed:
Update code/API_definitions/kyc-match.yaml

Fernando proposal: to keep -wip

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

ToshiWakayama-KDDI commented:
Hi @fernandopradocabrillo , HI @GillesInnov35 

I have committed additonal change requests and have just done Swagger check.  Please approve this PR, so that I can merge it.

Many thanks,
Toshi

ToshiWakayama-KDDI commented:
HI @fernandopradocabrillo , Hi @GillesInnov35 , 

Could you please approve this.

Thanks,
Toshi

ToshiWakayama-KDDI commented:
HI @fernandopradocabrillo , Hi @GillesInnov35 ,

Could you please approve this.

Thanks,
Toshi

fernandopradocabrillo reviewed (approved):
LGTM

fernandopradocabrillo commented:
Thanks @ToshiWakayama-KDDI for the effort

ToshiWakayama-KDDI commented:
Hi @fernandopradocabrillo , Hi @GillesInnov35 .

As my last comment above was at 02.20 am, I could not wait for Fernando's approval yesterday.  I have just merged the PR.  Thank you so much for your cooperation.  I will inform GSMA of this.

If you could approve the Fill-in one, PR #28, without delay, it would be much appreciated.  

Many thanks,
Toshi
